staging/kos/pkg/networkfabric/factory/factory.go

```diff
@@ -14,24 +14,28 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package localifc
+package networkfabric
 
 import (
-	"fmt"
-	"sync"
-)
+	"github.com/golang/glog"
 
-var registryMutex sync.Mutex
+	netfabric "k8s.io/examples/staging/kos/pkg/networkfabric"
+	"k8s.io/examples/staging/kos/pkg/networkfabric/types/logger"
+)
 
-var ifcs = make(map[string]Interface)
+const (
+	// TODO make logLvl settable from the outside (env variable, command line flag, etc...)
+	logLvl     = glog.Level(1)
+	loggerType = "logger"
+)
 
-func Register(name string, ifc Interface) error {
-	registryMutex.Lock()
-	defer func() { registryMutex.Unlock() }()
-	old := ifcs[name]
-	if old != nil && old != ifc {
-		return fmt.Errorf("Duplicate registration of local interface %q: %#+v and %#+v\n", name, old, ifc)
+func NewNetFabricForType(netFabricType string) netfabric.Interface {
```

Comment:
Rather than building the list of implementations into the factory file, use init() functions in other files so that each file register its own implementation.

Reply:
Fixed in a0cd7a2.